vulnerabilities/models.py

```diff
@@ -39,59 +39,87 @@ class Vulnerability(models.Model):
+        verbose_name_plural = "Vulnerabilities"
 
 
 class VulnerabilityReference(models.Model):
     """
     A reference to a vulnerability such as a security advisory from a Linux distribution or language
     package manager.
     """
-    vulnerability = models.ForeignKey(
-        Vulnerability, on_delete=models.CASCADE)
-    source = models.CharField(
-        max_length=50, help_text='Source(s) name eg:NVD', blank=True)
+
+    vulnerability = models.ForeignKey(Vulnerability, on_delete=models.CASCADE)
+    source = models.CharField(max_length=50, help_text="Source(s) name eg:NVD", blank=True)
     reference_id = models.CharField(
-        max_length=50, help_text='Reference ID, eg:DSA-4465-1', blank=True)
-    url = models.URLField(
-        max_length=1024, help_text='URL of Vulnerability data', blank=True)
+        max_length=50, help_text="Reference ID, eg:DSA-4465-1", blank=True
+    )
+    url = models.URLField(max_length=1024, help_text="URL of Vulnerability data", blank=True)
 
     class Meta:
-        unique_together = ('vulnerability', 'source', 'reference_id', 'url')
+        unique_together = ("vulnerability", "source", "reference_id", "url")
 
     def __str__(self):
-        return f'{self.source} {self.reference_id} {self.url}'
+        return f"{self.source} {self.reference_id} {self.url}"
 
 
 class Package(PackageURLMixin):
     """
     A software package with links to relevant vulnerabilities.
     """
+
     vulnerabilities = models.ManyToManyField(
-        to='Vulnerability', through='PackageRelatedVulnerability')
+        to="Vulnerability", through="PackageRelatedVulnerability"
+    )
+
+    @property
+    def vulnerable_to(self):
+        qs = PackageRelatedVulnerability.objects.filter(
+            package_id=self.id, is_vulnerable=True
+        ).select_related("vulnerability")
+        return [rel.vulnerability for rel in qs]
+
+    @property
+    def resolved_to(self):
+        qs = PackageRelatedVulnerability.objects.filter(
+            package_id=self.id, is_vulnerable=False
+        ).select_related("vulnerability")
+        return [rel.vulnerability for rel in qs]
 
     class Meta:
-        unique_together = ('name', 'namespace', 'type',
-                           'version', 'qualifiers', 'subpath')
+        unique_together = ("name", "namespace", "type", "version", "qualifiers", "subpath")
+
     # Remove the `qualifers` and `set_package_url` overrides after
     # https://github.com/package-url/packageurl-python/pull/35 gets merged
     qualifiers = pgfields.JSONField(
         default=dict,
         help_text=_(
-            'Extra qualifying data for a package such as the name of an OS, '
-            'architecture, distro, etc.'
+            "Extra qualifying data for a package such as the name of an OS, "
+            "architecture, distro, etc."
         ),
-        null=True
+        null=True,
     )
 
     def set_package_url(self, package_url):
```

Comment:
Why are we overriding the default method from PackageURLMixin?

Reply:
This one uses JSONField the old`set_package_url` converts qualifiers into normalized string. More on this https://github.com/package-url/packageurl-python/pull/35#discussion_r434320271